src/PrestaShopBundle/Form/Admin/Sell/Discount/DiscountConditionsType.php

```diff
@@ -78,7 +82,20 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 'row_attr' => [
                     'class' => 'discount-container minimum-amount-container',
                 ],
-            ]);
+            ])
+            ->add('specific_products', ProductSearchType::class, [
+                'layout' => EntitySearchInputType::LIST_LAYOUT,
+                'entry_type' => PackedProductType::class,
+                'filters' => [
+                    'filteredTypes' => [ProductType::TYPE_PACK],
+                ],
+                'limit' => 0,
+                'label' => 'Specific product',
+                'include_combinations' => true,
```

Comment:
```suggestion
                'include_combinations' => false,
```
For now focus on the products only, the `CartRule` is not able to handle a list of combinations yet so this should be handled in a different dedicated PR. Even the UX proposed is not about searching for combinations, it's a two steps process first search for a product, then select combinations from that specific product so it doesn't match the current possibilities of this component.